pkg/koordlet/runtimehooks/hooks/hooks.go

```diff
@@ -72,15 +72,21 @@ func getHooksByStage(stage rmconfig.RuntimeHookType) []*Hook {
 	}
 }
 
-func RunHooks(stage rmconfig.RuntimeHookType, protocol protocol.HooksProtocol) {
+func RunHooks(failPolicy rmconfig.FailurePolicyType, stage rmconfig.RuntimeHookType, protocol protocol.HooksProtocol) error {
 	hooks := getHooksByStage(stage)
 	klog.V(5).Infof("start run %v hooks at %s", len(hooks), stage)
 	for _, hook := range hooks {
 		klog.V(5).Infof("call hook %v", hook.name)
 		if err := hook.fn(protocol); err != nil {
-			klog.Warningf("failed to run hook %s in stage %s, reason: %v", hook.name, stage, err)
+			if failPolicy == rmconfig.PolicyFail {
+				klog.Errorf("failed to run hook %s in stage %s, reason: %v", hook.name, stage, err)
```

Comment:
rmconfig has been registered to RuntimeProxy,  so there is no need to filter errors basing on  rmconfig.PolicyFail duplicately. 
'return err' is enough.

Reply:
emm.. , for koordlet, under PolicyFail scenario, return error immediately when error happens, while PolicyIgnore, all hooks should be called even if errors happen for one or more hook?
if it is, it makes sense to check rmconfig.PolicyFail here.
but there had better be more comments here to avoid confusing.

Reply:
+1, if we really need to support `stop running other hooks once someone failed`, it is better to add another option choice to runtime hook in koordlet, since the `rmconfig.Policy` is only used for `runtime-proxy`.

Reply:
hi @huiwq1990, do you have any other ideas about these comments?

Reply:
@zwzhang0107 I agress the comments, and define a new options `RuntimeHooksCallFailurePolicy`. And just reuse the  `runtimeproxy.FailurePolicyType`.